### PR TITLE
Reflection_oM: Adding the IOutputObejct interface

### LINKED ISSUE #893: Reflection_oM: Add new interface for defining objects to be automatically exploded and never show up in the UI
#### Description:

To more easily handle methods returning multiple outputs, in a more manageable format for the toolkit, suggesting adding a new interface that can be attached to an object as `IOutputObject`. 

Any method returning this object should make it automatically explode in the UI, and display all of its properties, including their descriptions.

Any object implementing this interface should be filtered out from autocontructor and type methods in the UI.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #893

- Adding interface to be applied to objects that when returned from a method should be automatically exploded.

### CODE REVIEW

File: Reflection_oM/Interface/IOutputObject.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.ComponentModel;
+
+namespace BH.oM.Reflection.Interface
+{
+    [Description("Interface to be attached to obejcts that are to be automatically exploded when returned in any UI. Objects implementing this interface will not be constructable from the UI.")]
```
Comment (FraserGreenroyd) [inline Reflection_oM/Interface/IOutputObject.cs:27]: Sp. `obejcts` should be `objects`.

**Comment by IsakNaslundBh** [comment]
> Is this instead of #892 ? Or as well as?

Could be both, but for the use case needed, would be instead. This is mostly a prototype to test things out. Will add appropriate tags.

**Comment by FraserGreenroyd** [comment]
> > Is this instead of #892 ? Or as well as?
> 
> Could be both, but for the use case needed, would be instead. This is mostly a prototype to test things out. Will add appropriate tags.

Ok makes sense. Yeah probably good to be WIP cause I can see a few framework related questions arising out of this :smile:

**Comment by IsakNaslundBh** [comment]
> > > Is this instead of #892 ? Or as well as?
> > 
> > 
> > Could be both, but for the use case needed, would be instead. This is mostly a prototype to test things out. Will add appropriate tags.
> 
> Ok makes sense. Yeah probably good to be WIP cause I can see a few framework related questions arising out of this 😄

yep, agreed. Have tried tackle them in the adjoining PRs in the engine and UI, but not sure it is the most elegant/best way to handle it.

**Comment by adecler** [comment]
I have to admit I am a bit unsure about this one. I understand the idea behind it but I wonder if it isn't still better to output a single object that can then be exploded. We can for example make the analogy to the `IResult` objects. Those could easily be considered as `IOutputObject` and I would much prefer to keep them as they are.

**Comment by IsakNaslundBh** [comment]
Closing this PR as development and discussions have moved on. Leaving issue open, with a parked label on it for further reference.

**Comment by adecler** [comment]
@IsakNaslundBh , was that intentional that you didn't delete the branch ?

**Comment by IsakNaslundBh** [comment]
Was not, thanks!

**Pull request closed without merging**